packaging/version.py

```diff
@@ -257,8 +257,18 @@ def _legacy_cmpkey(version: str) -> LegacyCmpKey:
 class Version(_BaseVersion):
 
     _regex = re.compile(r"^\s*" + VERSION_PATTERN + r"\s*$", re.VERBOSE | re.IGNORECASE)
+    _obj_cache: Dict[str, "Version"] = {}
+
+    def __new__(cls, version: str) -> "Version":
```

Comment:
Why not use [`functools.lru_cache()`](https://docs.python.org/3/library/functools.html#functools.lru_cache)? And what about subclasses and superclasses which use the same `version` value?

Reply:
re: lru_cache, the bad answer is that mypy is really not ok with that.

I don't see why superclasses would come into play here?

re: subclasses - I don't immediately know the best approach